src/parquet/schema/descriptor.cc

```diff
@@ -58,19 +61,23 @@ void SchemaDescriptor::BuildTree(const NodePtr& node, int16_t max_def_level,
   // Now, walk the schema and create a ColumnDescriptor for each leaf node
   if (node->is_group()) {
     const GroupNode* group = static_cast<const GroupNode*>(node.get());
+    std::shared_ptr<ColumnPath> group_path = column_path->extend(group->name());
     for (int i = 0; i < group->field_count(); ++i) {
-      BuildTree(group->field(i), max_def_level, max_rep_level);
+      BuildTree(group->field(i), max_def_level, max_rep_level, group_path);
     }
   } else {
     // Primitive node, append to leaves
-    leaves_.push_back(ColumnDescriptor(node, max_def_level, max_rep_level, this));
+    leaves_.push_back(ColumnDescriptor(node, max_def_level, max_rep_level,
+      column_path->extend(node->name()), this));
```

Comment:
My preference is to add a `GetColumnPath` to `ColumnDescriptor` and build this on demand.